server/service/src/invoice/outbound_return/insert/mod.rs

```diff
@@ -0,0 +1,377 @@
+                    ctx,
+                    InsertStockOutLine {
+                        id: line.id.clone(),
+                        invoice_id: new_invoice.id.clone(),
+                        stock_line_id: line.stock_line_id.clone(),
+                        number_of_packs: line.number_of_packs.clone(),
+                        note: Some(line.note.clone()),
+                        r#type: Some(StockOutType::OutboundReturn),
+                        tax: None,
+                        total_before_tax: None,
+                    },
+                )
+                .map_err(|error| OutError::LineInsertError {
+                    line_id: line.id.clone(),
+                    error,
+                })?;
+
+                invoice_line_repo
+                    .update_return_reason_id(&line.id, Some(line.reason_id.clone()))
+                    .map_err(|error| OutError::LineReturnReasonUpdateError {
+                        line_id: line.id.clone(),
+                        error,
+                    })?;
+            }
+
+            activity_log_entry(
+                &ctx,
+                ActivityLogType::InvoiceCreated,
+                Some(new_invoice.id.to_owned()),
+                None,
+                None,
+            )?;
+
+            get_invoice(ctx, None, &new_invoice.id)
+                .map_err(|error| OutError::DatabaseError(error))?
+                .ok_or(OutError::NewlyCreatedInvoiceDoesNotExist)
+        })
+        .map_err(|error| error.to_inner_error())?;
+
+    Ok(outbound_return)
+}
+
+impl From<RepositoryError> for OutError {
+    fn from(error: RepositoryError) -> Self {
+        OutError::DatabaseError(error)
+    }
+}
+
+impl From<TransactionError<OutError>> for OutError {
+    fn from(error: TransactionError<OutError>) -> Self {
+        match error {
+            TransactionError::Transaction { msg, level } => {
+                OutError::DatabaseError(RepositoryError::TransactionError { msg, level })
+            }
+            TransactionError::Inner(e) => e,
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use repository::{
+        mock::{
+            mock_name_a, mock_outbound_return_a, mock_stock_line_b, mock_store_a,
+            mock_user_account_a, MockData, MockDataInserts,
+        },
+        test_db::setup_all_with_data,
+        InvoiceLineRowRepository, InvoiceRowRepository, NameRow, NameStoreJoinRow, RepositoryError,
+        ReturnReasonRow,
+    };
+    use util::{inline_edit, inline_init};
+
+    use crate::{
+        invoice::outbound_return::insert::{
+            InsertOutboundReturn, InsertOutboundReturnError as ServiceError,
+        },
+        invoice_line::stock_out_line::InsertStockOutLineError,
+        service_provider::ServiceProvider,
+    };
+
+    use super::InsertOutboundReturnLine;
+
+    #[actix_rt::test]
+    async fn test_insert_outbound_return_errors() {
+        fn not_visible() -> NameRow {
+            inline_init(|r: &mut NameRow| {
```

Comment:
Saw the comment about preferring `..Default::default()` over `inline_init` now after writing these... should I update?

Reply:
inline_init is fine still, the preference is ..Default::default() though, easier to understand and read